torch/_numpy/random.py

```diff
@@ -29,25 +29,48 @@
     "randint",
     "shuffle",
     "uniform",
-    "USE_NUMPY_RANDOM",
 ]
 
 
-USE_NUMPY_RANDOM = False
+def use_numpy_random():
+    # local import to avoid ref cycles
+    import torch._dynamo.config as config
+
+    return config.use_numpy_random_stream
 
 
 def deco_stream(func):
     @functools.wraps(func)
     def inner(*args, **kwds):
-        if USE_NUMPY_RANDOM is False:
+        if use_numpy_random() is False:
             return func(*args, **kwds)
-        elif USE_NUMPY_RANDOM is True:
-            from numpy import random as nr
+        elif use_numpy_random() is True:
+            import numpy
+
+            from ._ndarray import ndarray
+
+            f = getattr(numpy.random, func.__name__)
+
+            # numpy funcs accept numpy ndarrays, unwrap
+            args = tuple(
+                arg.tensor.numpy() if isinstance(arg, ndarray) else arg for arg in args
+            )
+            kwds = {
+                key: val.tensor.numpy() if isinstance(val, ndarray) else val
+                for key, val in kwds.items()
+            }
+
+            value = f(*args, **kwds)
+
+            # `value` can be either numpy.ndarray or python scalar (or None)
+            if isinstance(value, numpy.ndarray):
+                value = ndarray(torch.as_tensor(value))
 
-            f = getattr(nr, func.__name__)
-            return f(*args, **kwds)
+            return value
         else:
-            raise ValueError(f"USE_NUMPY_RANDOM={USE_NUMPY_RANDOM} not understood.")
+            raise ValueError(
+                f"config.use_numpy_random_stream = {use_numpy_random()} not understood."
+            )
```

Comment:
Not needed anymore. We expect people to respect the python types. Same above,  `is True / is False` is not necessary.